bindata/etcd/cluster-restore.sh

```diff
@@ -53,6 +53,34 @@ function restore_static_pods() {
     tar -xvf "${BACKUP_FILE}" --strip-components=2 -C "${MANIFEST_DIR}"/ "${BACKUP_POD_PATH}"
   done
 }
+function replace_node_specific_parameters() {
+  pod_file="$1"
+# shellcheck disable=SC2016
+  declare -A SUBS_MAP
+  SUBS_MAP['export ETCD_NAME=.*$']='export ETCD_NAME=${NODE_NODE_ENVVAR_NAME_ETCD_NAME}'
+  SUBS_MAP['export ETCD_INITIAL_CLUSTER=.*$']='export ETCD_INITIAL_CLUSTER="${ETCD_NAME}=https://${NODE_NODE_ENVVAR_NAME_ETCD_URL_HOST}:2380"'
+  SUBS_MAP['export ETCD_NODE_PEER_URL=.*$']='export ETCD_NODE_PEER_URL=https://${NODE_NODE_ENVVAR_NAME_ETCD_URL_HOST}:2380'
+  SUBS_MAP['--initial-advertise-peer-urls .*$']='--initial-advertise-peer-urls $ETCD_NODE_PEER_URL \\'
+  SUBS_MAP['--initial-advertise-peer-urls=.*$']='--initial-advertise-peer-urls=https://${NODE_NODE_ENVVAR_NAME_IP}:2380 \\'
+  SUBS_MAP['--cert-file=.*$']='--cert-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-serving/etcd-serving-NODE_NAME.crt \\'
+  SUBS_MAP['--key-file=.*$']='--key-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-serving/etcd-serving-NODE_NAME.key \\'
+  SUBS_MAP['--peer-cert-file=.*$']='--peer-cert-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-peer/etcd-peer-NODE_NAME.crt \\'
+  SUBS_MAP['--peer-key-file=.*$']='--peer-key-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-peer/etcd-peer-NODE_NAME.key \\'
+  SUBS_MAP['--advertise-client-urls=.*$']='--advertise-client-urls=https://${NODE_NODE_ENVVAR_NAME_IP}:2379 \\'
+
+  for K in "${!SUBS_MAP[@]}"
+  do
+    # check if the pattern exists, and return with error if it doesn't
+    if ! grep -- "$K" "$pod_file" > /dev/null 2>&1
+    then
```

Comment:
again style `if ! stuff ; then`

Reply:
I honestly don't feel strongly, either way, it just should be standardized in the script